### PR TITLE
Add error info if no user can fork the repo

### PR DESCRIPTION
**Pull request by yp05327**
Before: (Owner list is empty)
image

After: (Disable the button and show the error info)
image

### CODE REVIEW

**Comment by silverwind** [comment]
"This repository can not be forked" would sound better. Can we also give some hint on the reason why it can not be done?

File: options/locale/locale_en-US.ini [hunk 1]
```diff
@@ -935,6 +935,7 @@ fork_from = Fork From
 already_forked = You've already forked %s
 fork_to_different_account = Fork to a different account
 fork_visibility_helper = The visibility of a forked repository cannot be changed.
+no_fork_user = No user can fork this repo.
```
Comment (silverwind) [inline options/locale/locale_en-US.ini]:
```suggestion
fork_not_possible = This repository can not be forked.
```

File: routers/web/repo/pull.go [hunk 2]
```diff
@@ -174,6 +174,12 @@ func getForkRepository(ctx *context.Context) *repo_model.Repository {
 		ctx.Data["ContextUser"] = ctx.Doer
 	} else if len(orgs) > 0 {
 		ctx.Data["ContextUser"] = orgs[0]
+	} else {
+		msg := ctx.Tr("repo.no_fork_user")
```
Comment (silverwind) [inline routers/web/repo/pull.go]:
```suggestion
		msg := ctx.Tr("repo.fork_not_possible")
```

**Comment by yp05327** [comment]
> Can we also give some hint on the reason why it can not be done?

Updated the error info, does it look good to you?

**Review by silverwind — APPROVED** [review]

**Comment by silverwind** [comment]
Looks good, but could also be made a single translation string I guess as I'm not sure whether `<text>: <text>` constructs are the same in all languages.

**Comment by yp05327** [comment]
> Looks good, but could also be made a single translation string I guess as I'm not sure whether `<text>: <text>` constructs are the same in all languages. If we combine it, use `This repository can not be forked because there are no valid owners.`.

Updated. 😄

**Review by silverwind — APPROVED** [review]

**Review by puni9869 — APPROVED** [review]

File: routers/web/repo/pull.go [hunk 3]
```diff
@@ -174,6 +174,12 @@ func getForkRepository(ctx *context.Context) *repo_model.Repository {
 		ctx.Data["ContextUser"] = ctx.Doer
 	} else if len(orgs) > 0 {
 		ctx.Data["ContextUser"] = orgs[0]
+	} else {
+		msg := ctx.Tr("repo.fork_no_valid_owners")
+		ctx.Data["Flash"] = ctx.Flash
+		ctx.Flash.Error(msg)
```
Comment (wxiaoguang) [inline routers/web/repo/pull.go:180]: It would break the "Flash" framework, it should be `ctx.Flash.Error(msg, true)`

Comment (wxiaoguang) [reply routers/web/repo/pull.go:180]: ->  Fix "Flash" message usage #25895

**Pull request merged**